src/sentry/rules/actions/notify_event_service.py

```diff
@@ -3,16 +3,85 @@
 """
 from __future__ import absolute_import
 
+import logging
 from django import forms
 
+from sentry.api.serializers import serialize
+from sentry.api.serializers.models.app_platform_event import AppPlatformEvent
+from sentry.api.serializers.models.incident import IncidentSerializer
+from sentry.constants import SentryAppInstallationStatus
 from sentry.incidents.logic import get_alertable_sentry_apps
+from sentry.incidents.models import INCIDENT_STATUS, IncidentStatus
+from sentry.integrations.metric_alerts import incident_attachment_info
+from sentry.models import SentryApp, SentryAppInstallation
 from sentry.plugins.base import plugins
 from sentry.rules.actions.base import EventAction
 from sentry.rules.actions.services import PluginService, SentryAppService
-from sentry.models import SentryApp
-from sentry.utils.safe import safe_execute
+from sentry.tasks.sentry_apps import notify_sentry_app, send_and_save_webhook_request
 from sentry.utils import metrics
-from sentry.tasks.sentry_apps import notify_sentry_app
+from sentry.utils.safe import safe_execute
+
+
+logger = logging.getLogger("sentry.integrations.sentry_app")
+
+
+def build_incident_attachment(incident, metric_value=None):
+    from sentry.api.serializers.rest_framework.base import (
+        camel_to_snake_case,
+        convert_dict_key_case,
+    )
+
+    data = incident_attachment_info(incident, metric_value)
+    return {
+        "metric_alert": convert_dict_key_case(
+            serialize(incident, serializer=IncidentSerializer()), camel_to_snake_case
+        ),
+        "text": data["text"],
+        "title": data["title"],
```

Comment:
This is the title of the rule yes? I think it might be good to be more specific with the names of these attributes, like `rule_title`, and `rule_web_url` (assuming its the web url and not the api), and is text the condition? 

You don't have to have those names perse but something a little more descriptive would be good